control-plane/api-gateway/binding/result.go

```diff
@@ -382,59 +383,75 @@ func (l listenerValidationResult) conflictedCondition(generation int64) metav1.C
-	case errListenerInvalidCertificateRef_NotFound, errListenerInvalidCertificateRef_NotSupported, errListenerInvalidCertificateRef_InvalidData, errListenerInvalidCertificateRef_NonFIPSRSAKeyLen, errListenerInvalidCertificateRef_FIPSRSAKeyLen:
-		return metav1.Condition{
-			Type:               "ResolvedRefs",
-			Status:             metav1.ConditionFalse,
-			Reason:             "InvalidCertificateRef",
-			ObservedGeneration: generation,
-			Message:            l.refErr.Error(),
-			LastTransitionTime: now,
-		}
-	case errRefNotPermitted:
-		return metav1.Condition{
-			Type:               "ResolvedRefs",
-			Status:             metav1.ConditionFalse,
-			Reason:             "RefNotPermitted",
-			ObservedGeneration: generation,
-			Message:            l.refErr.Error(),
-			LastTransitionTime: now,
+	for _, refErr := range l.refErrs {
+		switch refErr {
+		case errListenerInvalidCertificateRef_NotFound, errListenerInvalidCertificateRef_NotSupported, errListenerInvalidCertificateRef_InvalidData, errListenerInvalidCertificateRef_NonFIPSRSAKeyLen, errListenerInvalidCertificateRef_FIPSRSAKeyLen:
+			conditions = append(conditions, metav1.Condition{
+				Type:               "ResolvedRefs",
+				Status:             metav1.ConditionFalse,
+				Reason:             "InvalidCertificateRef",
+				ObservedGeneration: generation,
+				Message:            refErr.Error(),
+				LastTransitionTime: now,
+			})
+		case errListenerJWTProviderNotFound:
+			conditions = append(conditions, metav1.Condition{
+				Type:               "ResolvedRefs",
+				Status:             metav1.ConditionFalse,
+				Reason:             "InvalidJWTProviderRef",
+				ObservedGeneration: generation,
+				Message:            refErr.Error(),
+				LastTransitionTime: now,
+			})
+		case errRefNotPermitted:
+			conditions = append(conditions, metav1.Condition{
+				Type:               "ResolvedRefs",
+				Status:             metav1.ConditionFalse,
+				Reason:             "RefNotPermitted",
+				ObservedGeneration: generation,
+				Message:            refErr.Error(),
+				LastTransitionTime: now,
+			})
 		}
-	default:
-		return metav1.Condition{
+	}
+	if len(conditions) == 0 {
+		conditions = append(conditions, metav1.Condition{
 			Type:               "ResolvedRefs",
 			Status:             metav1.ConditionTrue,
 			Reason:             "ResolvedRefs",
 			ObservedGeneration: generation,
 			Message:            "resolved certificate references",
 			LastTransitionTime: now,
-		}
+		})
 	}
+	return conditions
 }
 
 // Conditions constructs the entire set of conditions for a given gateway listener.
 func (l listenerValidationResult) Conditions(generation int64) []metav1.Condition {
-	return []metav1.Condition{
+	conditions := []metav1.Condition{
 		l.acceptedCondition(generation),
 		l.programmedCondition(generation),
 		l.conflictedCondition(generation),
-		l.resolvedRefsCondition(generation),
 	}
+	conditions = append(conditions, l.resolvedRefsConditions(generation)...)
+	return conditions
```

Comment:
You can cut a line here 🤷🏻 
```suggestion
	return append(conditions, l.resolvedRefsConditions(generation)...)
```